src/devices/bus/ieee488/remote488.cpp

```diff
@@ -0,0 +1,720 @@
+    sees.  Connection status is set to "down" when three consecutive J
+    msgs are sent out without being replied to.
+
+    - P & Q messages
+
+    The 'P' msg is used by an external device to set the response to
+    be placed on the bus when a local controller does a parallel
+    poll. Message byte is encoded in positive logic (i.e. any bit set
+    to 1 is forced to 0 on data bus during parallel poll).  The
+    response set by P msg is used during parallel polls until changed
+    by another P msg. A P msg is never sent out by the remotizer.
+    The Q msg is sent by the remotizer whenever a parallel poll is
+    performed locally. Its purpose is to solicit a P msg from the
+    external device. The external device may choose to send P msgs in
+    reply to Q msgs or to ignore them and send P msgs asynchronously.
+    Parallel poll is usually a very fast operation (a few microseconds)
+    so it's possible, due to latency, that the reply to a Q msg is
+    applied on the bus many polls later. The remotizer doesn't repeat
+    the transmission of a Q msg until a P msg is received to avoid
+    generating a lot of traffic if the local controller rapidly
+    repeats parallel polls.
+
+    - R & S messages
+
+    These messages are used to align the state of bus control lines
+    with the external devices. Each control line is mapped to a bit in
+    the accompanying byte, according to this table.
+
+    | Bit | Signal |
+    |-----+--------|
+    | 0   | ATN    |
+    | 1   | IFC    |
+    | 2   | REN    |
+    | 3   | SRQ    |
+
+    A "R" message clears (sets to 0, i.e. asserts) all signals that
+    are set to "1" in the byte, all others retain their value.
+    A "S" message sets signals to 1, i.e. de-asserts them.
+    The remotizer may respond with a "R" message if an incoming "S"
+    message is attempting to set to 1 a signal that is locally forced
+    to 0.
+
+    Limits & potential issues:
+    - There are no checks for violations of IEEE-488 protocol. It's
+      possible, for example, to put the bus in an illegal state by
+      sending a 'E' msg when the local controller is asserting ATN
+      (thus creating a phantom parallel poll).
+    - The unacknowledged nature of D/E msgs may cause problems if
+      the sender of the msgs must be synchronized byte-for-byte with
+      the receiver. There is no direct way, for example, for the
+      sender of a string of bytes to know how many have been accepted
+      by the receiver.
+    - It's difficult to achieve accurate synchronization between the
+      local emulation time and the external time.
+
+    TODOs/possible enhancements:
+    - Implement handling of incoming Q msgs (needed when parallel poll
+      is being performed by a remote controller)
+    - Enhancement: implement a msg for accurate time synchronization
+
+*********************************************************************/
+
+#include "remote488.h"
```

Comment:
You need to `#include "emu.h"` as the very first thing apart from comments in a compilation unit.